src/services/workflow.service.js

```diff
@@ -0,0 +1,176 @@
+        return {};
+    }
+    for (let selection of a.selectionSet.selections) {
+        if (selection.kind === 'Field') {
+            selections[selection.name.value] = selection;
+        }
+    }
+    return selections;
+}
+
+
+function mergeSelection(a, b) {
+    /* merge b into a */
+    var aSel = getSelections(a);
+    var bSel = getSelections(b);
+    for (let selection in bSel) {
+        if (!(selection in aSel)) {
+            a.selectionSet.selections.push(gClone(bSel[selection]));
+        } else {
+            mergeSelection(aSel[selection], bSel[selection]);
+        }
+    }
+}
+
+
+function merge(a, b) {
+    /* merge two graphql schema */
+    mergeSelection(a.definitions[0], b.definitions[0]);
+}
+
+
+function prefix_lines(str, pref) {
+    return pref + str.replace(/\n/g, `\n${pref}`);
+}
+
+
+class WorkflowService {
+
+  constructor() {
+    this.apolloClient = createApolloClient(
+        `${window.location.pathname}/graphql`
+    );
+    //this.transformer = new VueAdsTableTreeTransformer()
+    this.query;
+    this.subscriptions = [];
+    this.views = new WeakMap();
+
+    this.polling = setInterval(() => {
+      this.request();
+    }, 5000)
+  }
+
+  destructor() {
+    clearInterval(this.polling);
+  }
+
+  getHash() {
+    /* psudo-random hash for internal identification only */
+    return Math.random();
+  }
+
+  recompute () {
+    /* recompute the combined query */
+    if (this.subscriptions.length < 1) {
+        this.query = null;
+        return;
+    }
+    // merge together active subscriptions
+    var root = gClone(this.subscriptions[0][2]);
+    for (let subscription of this.subscriptions.slice(1)) {
+        merge(root, subscription[2]);
+    }
+    this.query = root;
+
+    // TODO - remove
+    this.print();
+  }
+
+  register(view) {
+    /* register a new view */
+    this.views[view.viewId] = view
+  }
+
+  unregister(view) {
+    /* unregister a view (all subscriptions will be dropped */
+    this.view.remove(view);
+    this.subscriptions = this.subscriptions.filter(
+      item => item[1] != view
+    );
+    this.recompute();
+  }
+
+  subscribe(view, query) {
+    /* subscribe a view to a query */
+    var hash = this.getHash();
+    this.subscriptions.push(
+        [hash, view, parse(query)]
+    );
+    this.recompute();
+    return hash;
+  }
+
+  unsubscribe (hash) {
+    /* un-subscribe a view from a query */
+    this.subscriptions = this.subscriptions.filter(
+      item => item[0] != hash
+    );
+      this.recompute();
+  }
+
+  request() {
+    console.log('graphql:', this.query);
+
+    this.apolloClient.query({
+      query: this.query,
+      fetchPolicy: 'no-cache'
+    }).then((response) => {
+      const workflows = response.data.workflows;
+      store.dispatch('workflows/set', workflows);
+      return true;
+    }).catch((error) => {
+      const alert = new Alert(error.message, null, 'error');
+      store.dispatch('addAlert', alert);
```

Comment:
s/addAlert/setAlert, as we now have a single alert (it should be logged to browser console, and the latest alert will be displayed in the UI)